### ISSUE TITLE
Issue installing from git+ssh

### ISSUE THREAD

**Issue by vernak2539** [yarnpkg/yarn#3608]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
We get a failure during install, which seems to be related to the `git+ssh` paths

**_Output (initial run, not verbose)._** 
You can see it collects the changes correctly from the git repos, so it has access
```
[11:15:54][Step 2/3] yarn install --ignore-engines (54s)
[11:15:55][yarn install --ignore-engines] yarn install v0.24.6
[11:15:55][yarn install --ignore-engines] [1/4] Resolving packages...
[11:15:56][yarn install --ignore-engines] [2/4] Fetching packages...
[11:16:49][yarn install --ignore-engines] info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
[11:16:49][yarn install --ignore-engines] error Command failed.
[11:16:49][yarn install --ignore-engines] Exit code: 1
[11:16:49][yarn install --ignore-engines] Command: git
[11:16:49][yarn install --ignore-engines] Arguments: pull
[11:16:49][yarn install --ignore-engines] Directory: /home/teamcity/.cache/yarn/v1/.tmp/11c50f8c3055ece490b38d4093c0e50f
[11:16:49][yarn install --ignore-engines] Output:
[11:16:49][yarn install --ignore-engines] Failed to add the host to the list of known hosts (/home/teamcity/.ssh/known_hosts).
[11:16:49][yarn install --ignore-engines] From ssh://github.something.com/uxarchitecture/fuelux-lightning-theme
[11:16:49][yarn install --ignore-engines]  * [new branch]      GH26---deal-with-bower-install -> origin/GH26---deal-with-bower-install
[11:16:49][yarn install --ignore-engines]  * [new branch]      cmcculloh-GH45---wizard-should-look-like-the-slds-wizard_or_fuelux-lightning-theme -> origin/cmcculloh-GH45---wizard-should-look-like-the-slds-wizard_or_fuelux-lightning-theme
[11:16:49][yarn install --ignore-engines]  * [new branch]      cmcculloh-GH70---wizard-step-content-to-utilize-design-tokens -> origin/cmcculloh-GH70---wizard-step-content-to-utilize-design-tokens
[11:16:49][yarn install --ignore-engines]  * [new branch]      fix-alert-icons-hidden-by-new-non-white-defaults-for-success-and-danger -> origin/fix-alert-icons-hidden-by-new-non-white-defaults-for-success-and-danger
[11:16:49][yarn install --ignore-engines]  * [new branch]      jamin-branch -> origin/jamin-branch
[11:16:49][yarn install --ignore-engines]  * [new branch]      master     -> origin/master
[11:16:49][yarn install --ignore-engines]  * [new branch]      testing-release-process -> origin/testing-release-process
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.2.2      -> 0.2.2
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.6.2      -> 0.6.2
[11:16:49][yarn install --ignore-engines] From ssh://github.something.com/uxarchitecture/fuelux-lightning-theme
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.2.0      -> 0.2.0
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.2.3      -> 0.2.3
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.2.4      -> 0.2.4
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.2.5      -> 0.2.5
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.3.0      -> 0.3.0
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.3.1      -> 0.3.1
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.3.2      -> 0.3.2
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.4.0      -> 0.4.0
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.4.1      -> 0.4.1
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.4.2      -> 0.4.2
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.4.3      -> 0.4.3
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.4.4      -> 0.4.4
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.5.0      -> 0.5.0
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.5.1      -> 0.5.1
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.5.2      -> 0.5.2
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.6.0      -> 0.6.0
[11:16:49][yarn install --ignore-engines]  * [new tag]         0.6.1      -> 0.6.1
[11:16:49][yarn install --ignore-engines] There is no tracking information for the current branch.
[11:16:49][yarn install --ignore-engines] Please specify which branch you want to merge with.
[11:16:49][yarn install --ignore-engines] See git-pull(1) for details
[11:16:49][yarn install --ignore-engines] 
[11:16:49][yarn install --ignore-engines]     git pull <remote> <branch>
[11:16:49][yarn install --ignore-engines] 
[11:16:49][yarn install --ignore-engines] If you wish to set tracking information for this branch you can do so with:
[11:16:49][yarn install --ignore-engines] 
[11:16:49][yarn install --ignore-engines]     git branch --set-upstream-to=origin/<branch> master
[11:16:49][yarn install --ignore-engines] Process exited with code 1
```

**_Output  w/ verbose_**
```
[11:27:09][Step 2/3] yarn install --ignore-engines --verbose (3s)
[11:27:09][yarn install --ignore-engines --verbose] yarn install v0.24.6
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.571 Checking for configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.572 Found configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.574 Checking for configuration file "/home/teamcity/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.574 Checking for configuration file "/opt/teamcity_agent/system/jonnyzzz.nvm/versions/node/v6.5.0/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.574 Checking for configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.574 Found configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.575 Checking for configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.575 Checking for configuration file "/opt/teamcity_agent/work/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.575 Checking for configuration file "/opt/teamcity_agent/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.576 Checking for configuration file "/opt/.npmrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.578 Checking for configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.578 Found configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.578 Checking for configuration file "/home/teamcity/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.579 Found configuration file "/home/teamcity/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.579 Checking for configuration file "/opt/teamcity_agent/system/jonnyzzz.nvm/versions/node/v6.5.0/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.579 Checking for configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.58 Found configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/client/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.58 Checking for configuration file "/opt/teamcity_agent/work/7beaa493b1ca74b5/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.58 Checking for configuration file "/opt/teamcity_agent/work/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.58 Checking for configuration file "/opt/teamcity_agent/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.58 Checking for configuration file "/opt/.yarnrc".
[11:27:09][yarn install --ignore-engines --verbose] verbose 0.583 current time: 2017-06-08T10:27:09.738Z
[11:27:09][yarn install --ignore-engines --verbose] [1/4] Resolving packages...
[11:27:10][yarn install --ignore-engines --verbose] [2/4] Fetching packages...
[11:27:12][yarn install --ignore-engines --verbose] verbose 3.663 Error: Command failed.
[11:27:12][yarn install --ignore-engines --verbose] Exit code: 1
[11:27:12][yarn install --ignore-engines --verbose] Command: git
[11:27:12][yarn install --ignore-engines --verbose] Arguments: pull
[11:27:12][yarn install --ignore-engines --verbose] Directory: /home/teamcity/.cache/yarn/v1/.tmp/11c50f8c3055ece490b38d4093c0e50f
[11:27:12][yarn install --ignore-engines --verbose] Output:
[11:27:12][yarn install --ignore-engines --verbose] There is no tracking information for the current branch.
[11:27:12][yarn install --ignore-engines --verbose] Please specify which branch you want to merge with.
[11:27:12][yarn install --ignore-engines --verbose] See git-pull(1) for details
[11:27:12][yarn install --ignore-engines --verbose] 
[11:27:12][yarn install --ignore-engines --verbose]     git pull <remote> <branch>
[11:27:12][yarn install --ignore-engines --verbose] 
[11:27:12][yarn install --ignore-engines --verbose] If you wish to set tracking information for this branch you can do so with:
[11:27:12][yarn install --ignore-engines --verbose] 
[11:27:12][yarn install --ignore-engines --verbose]     git branch --set-upstream-to=origin/<branch> master
[11:27:12][yarn install --ignore-engines --verbose]     at MessageError (/opt/yarn/lib/yarn-cli.js:138:5)
[11:27:12][yarn install --ignore-engines --verbose]     at SpawnError (/opt/yarn/lib/yarn-cli.js:148:1)
[11:27:12][yarn install --ignore-engines --verbose]     at ChildProcess.proc.on.code (/opt/yarn/lib/yarn-cli.js:34034:15)
[11:27:12][yarn install --ignore-engines --verbose]     at emitTwo (events.js:106:13)
[11:27:12][yarn install --ignore-engines --verbose]     at ChildProcess.emit (events.js:191:7)
[11:27:12][yarn install --ignore-engines --verbose]     at maybeClose (internal/child_process.js:877:16)
[11:27:12][yarn install --ignore-engines --verbose]     at Socket.<anonymous> (internal/child_process.js:334:11)
[11:27:12][yarn install --ignore-engines --verbose]     at emitOne (events.js:96:13)
[11:27:12][yarn install --ignore-engines --verbose]     at Socket.emit (events.js:188:7)
[11:27:12][yarn install --ignore-engines --verbose]     at Pipe._handle.close [as _onclose] (net.js:493:12)
[11:27:12][yarn install --ignore-engines --verbose] info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
[11:27:12][yarn install --ignore-engines --verbose] error Command failed.
[11:27:12][yarn install --ignore-engines --verbose] Exit code: 1
[11:27:12][yarn install --ignore-engines --verbose] Command: git
[11:27:12][yarn install --ignore-engines --verbose] Arguments: pull
[11:27:12][yarn install --ignore-engines --verbose] Directory: /home/teamcity/.cache/yarn/v1/.tmp/11c50f8c3055ece490b38d4093c0e50f
[11:27:12][yarn install --ignore-engines --verbose] Output:
[11:27:12][yarn install --ignore-engines --verbose] There is no tracking information for the current branch.
[11:27:12][yarn install --ignore-engines --verbose] Please specify which branch you want to merge with.
[11:27:12][yarn install --ignore-engines --verbose] See git-pull(1) for details
[11:27:12][yarn install --ignore-engines --verbose] 
[11:27:12][yarn install --ignore-engines --verbose]     git pull <remote> <branch>
[11:27:12][yarn install --ignore-engines --verbose] 
[11:27:12][yarn install --ignore-engines --verbose] If you wish to set tracking information for this branch you can do so with:
[11:27:12][yarn install --ignore-engines --verbose] 
[11:27:12][yarn install --ignore-engines --verbose]     git branch --set-upstream-to=origin/<branch> master
[11:27:12][yarn install --ignore-engines --verbose] Process exited with code 1
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Unfortunately this is private, but will include a likeness of the package.json. All we did was upgrade from **v0.21.3** to **v0.24.6** and we get this error. This is running on an TeamCity instance, which has all the same access as our local development, but it works locally on a Mac.

package.json (not working)
```
  "dependencies": {
    "ad-studio-menu-registrar": "^0.8.3",
    "backbone": "^1.3.3",
    "bluebird": "^3.4.1",
    "datalibrary": "git+ssh://git@github.something.com/RogueSquadron/data-library.git#master",
    "datamapper": "git+ssh://git@github.something.com/RogueSquadron/data-mapper.git#master",
    "dragula": "^3.7.2",
    "enlightenment": "git+ssh://git@github.something.com/RogueSquadron/enlightenment.git#master",
    "et-mc-lib": "git+ssh://git@github.something.com/MarketingCloudDev/mc-lib.git#1.6.6",
    "fuelux": "3.15.11",
    "fuelux-lightning-theme": "git+ssh://git@github.something.com/uxarchitecture/fuelux-lightning-theme.git#0.5.1",
    "fueluxcomponentcore": "git+ssh://git@github.something.com/RogueSquadron/fueluxcomponentcore.git#master",
    "handlebars": "^4.0.6",
    "highcharts": "^5.0.10",
    "jquery": "3.2.1",
    "moment": "2.18.1",
    "numeral": "^2.0.6",
    "postmonger": "0.0.16",
    "underscore": "^1.8.3",
    "urijs": "^1.18.4"
  },
```

**What is the expected behavior?**
The expected behaviour is to install the deps correctly. This worked on **v0.21.3**, but we have upgraded and it seems to have changed. We have tried upgrading the lockfiles with the new version. It still works on **v0.21.3** but NOT **v0.24.6**. 

The weird thing is we have it pulling from another package.json which just has one `ssh+git` in it's dependencies just fine

**Please mention your node.js, yarn and operating system version.**
node: 6.5.0
os: centos 7
yarn: working -> v0.21.3, not working -> v0.24.6

**Comment by vernak2539** [comment]
Even tried switching all `#master`'s to use tags. no dice

**Comment by vernak2539** [comment]
This seems to be related to https://github.com/yarnpkg/yarn/issues/1248#issuecomment-255855737

**Comment by chase** [comment]
@vernak2539 This should have been resolved with #2384

**Comment by vernak2539** [comment]
No idea why, or how, but it seems like it has been resolved. Thanks
